src/realm/sync/subscriptions.cpp

```diff
@@ -970,9 +964,9 @@ SubscriptionSet SubscriptionStore::get_by_version(int64_t version_id)
     throw KeyNotFound(util::format("Subscription set with version %1 not found", version_id));
 }
 
-SubscriptionSet SubscriptionStore::get_refreshed(ObjKey key, int64_t version, std::optional<DB::VersionID> db_version)
+SubscriptionSet SubscriptionStore::get_refreshed(ObjKey key, int64_t version, std::optional<TransactionRef> maybe_tr)
```

Comment:
Perhaps you don't need std::optional here. `TransactionRef` is already a pointer type that can be checked for null.

Reply:
I think its clearer with the optional.

Reply:
It's just that we have a tradition for checking the pointer directly.

Reply:
okay, done. can you take another look?